data/campaigns/Sceptre_of_Fire/scenarios/4_Gathering_Materials.cfg

```diff
@@ -588,6 +591,40 @@
         [/message]
     [/event]
 
+    [event]
+        name=turn 3
+
+        [fire_event]
+            name=reminder_to_recruit_miners
+        [/fire_event]
+    [/event]
+    [event]
+        name=recruit,recall
+        first_time_only=no
+        [filter]
+            side=1
+        [/filter]
+
+        [store_gold]
+            side=1
+            variable=money_for_recruitment
+        [/store_gold]
+        [if]
+            [variable]
+                name=money_for_recruitment
+                less_than=65
+                # Two miners cost 38, the most expensive recruit is 24, and
+                # upkeep will cost a bit, hence warn around about 65.
+            [/variable]
+            [then]
+                [fire_event]
+                    name=reminder_to_recruit_miners
+                [/fire_event]
+            [/then]
+        [/if]
+        {CLEAR_VARIABLE money_for_recruitment}
+    [/event]
+
```

Comment:
Note that this does prevent undo when recalling, which can be avoided here with an [allow_undo]. To allow undo even when the message is displayed (and I see no reason why a hint message should prevent undo), you **might** have to add one to reminder_to_recruit_miners as well, though.

Reply:
Completely agreed, I need to fix this before it's ready for merging. However, I might add more documentation to the Wiki and unit-tests for the engine before fixing this PR.

A quick test shows a logical but under-documented behaviour for [allow_undo] and [fire_event]. Without changing the sub-event:
* putting the [allow_undo] before the [fire_event] does not allow undoing
* putting the [allow_undo] after the [fire_event] allows undoing

Without the sub-event firing, [allow_undo] works in a `name=recruit,recall` event without needing to split the event. It allowing undoing recalls but doesn't allowing undoing recruits that triggered the RNG.